go/vt/vtgate/vindexes/vschema.go

```diff
@@ -374,11 +409,37 @@ func buildKeyspaceReferences(vschema *VSchema, ksvschema *KeyspaceSchema) error
 			)
 		}
 
-		// Validate that reference can be resolved.
-		_, sourceT, err := vschema.findKeyspaceAndTableBySource(source)
-		if sourceT == nil {
+		// Verify the reference source can be resolved.
+		sourceT, err := vschema.findTable(
+			sourceKsname,
+			sourceTname,
+			false, /* constructTableIfNotFound */
+		)
+		// Rephrase errors to be more helpful in the context of VSchema linting.
+		if err != nil {
+			switch vterrors.Code(err) {
```

Comment:
nitpick: we are trying to minimize the indentation in our code. these checks can be done using a exit early strategy instead:

```go
// Rephrase errors to be more helpful in the context of VSchema linting.
if err != nil {
	if vterrors.Code(err) != vtrpcpb.Code_NOT_FOUND || vterrors.ErrState(err) != vterrors.BadDb {
		return err
	}
	return vterrors.Errorf(
		vtrpcpb.Code_NOT_FOUND,
		"source %q references a non-existent keyspace %q",
		source,
		sourceKsname,
	)
}
```

Reply:
Less indentation _and_ fewer lines of code, 👍